app/src/org/commcare/android/resource/installers/ProfileAndroidInstaller.java

```diff
@@ -119,6 +128,29 @@ public boolean install(Resource r, ResourceLocation location, Reference ref,
         return false;
     }
 
+    private void storeReleasedTime(AndroidCommCarePlatform platform, Reference ref) {
+        long releasedOnTime = -1;
+        if (ref instanceof JavaHttpReference) {
+            Headers responseHeaders = ((JavaHttpReference)ref).getResponseHeaders();
```

Comment:
Slight nitpick / style point - It's a tradeoff of 'correctness' v. verbosity, but patterns like this kind of defeat the point of the interface/inheritance structure. 

I think the "Java" way to do this would be to make a new interface for References (like "MetadataSupportedReference") which instead of returning the headers raw would return the Released On Date directly, and then do (if ref instanceof MetadatasupportedReference) { ref.getReleasedOnDate()}

That keeps the HTTP-centric code in the HTTP centric class and lets another reference extend this same behavior without needing to modify the locations where the behavior is relied upon.

Reply:
This makes sense. Further Question though - Does this mean we should create another `ReferenceFactory` that instantiates this `MetadataSupportedReference` or use the current `JavaHttpRoot` that based on some condition (all profile urls ?) differentiate between `JavaHttpReference` and `MetadataSupportedReference` ?

Reply:
I don't think that we need to change anything other than the HttpReference to follow this pattern, the new metadata class would be something that's added directly to qualifying references, rather than something that changes how references work.

The old scenario was

`
interface ref {};
...
class href implements ref {}
...
if (r instanceof href) {}
`

the new one would be 

`
interface ref{};
...
interface ref_meta extends ref { getMetadata(); }
..
class href implements ref, ref_meta {
getMetadata() {};
}
...
if (r instanceof ref_meta) {}
`
So the only three files that change are the new Meta interface, the HttpReference itself, and this class